src/platform/packages/shared/kbn-unified-metrics-grid/src/components/metrics_grid.tsx

```diff
@@ -78,47 +66,73 @@ export const MetricsGrid = ({
 
   if (pivotOn === 'metric' && fields.length === 0) {
     return (
-      <EuiText textAlign="center" color="subdued">
-        No metrics found matching &quot;{searchTerm}&quot;
-      </EuiText>
+      <div
+        css={css`
+          width: 100%;
+          height: 100%;
+        `}
+      >
+        <EuiFlexGroup
+          direction="column"
+          alignItems="center"
+          justifyContent="spaceAround"
+          css={css`
+            height: 100%;
+          `}
+          gutterSize="s"
+        >
+          <EuiFlexItem
+            css={css`
+              justify-content: end;
+            `}
+          >
+            <EuiIcon type={IconChartBarStacked as IconType} size="l" />
+          </EuiFlexItem>
+          <EuiFlexItem>
+            <EuiText size="xs">
+              {i18n.translate('metricsExperience.grid.noData', {
+                defaultMessage: 'No results found',
+              })}
+            </EuiText>
+          </EuiFlexItem>
+        </EuiFlexGroup>
+      </div>
     );
   }
 
   return (
     <EuiFlexGrid
-      data-test-subj="metrics-grid"
-      columns={getColumns()}
-      gutterSize="l"
-      style={{ margin: '16px' }}
+      columns={columns}
+      gutterSize="s"
+      css={css`
+        overflow: auto;
+        padding: ${euiTheme.size.s} ${euiTheme.size.s} 0;
+        ${euiScrollBarStyles(euiThemeContext)}
+      `}
     >
       {pivotOn === 'metric'
         ? fields.map((field, index) => (
-            <EuiFlexItem
-              key={`${field.name}-${displayDensity}`}
-              data-test-subj={`metric-chart-${field.name}`}
-            >
+            <EuiFlexItem key={field.name}>
               <MetricChart
                 metric={field}
                 timeRange={timeRange}
                 dimensions={dimensions}
                 filters={filters}
                 colorIndex={index}
-                displayDensity={displayDensity}
-                data-test-subj={`metric-chart-${field.name}`}
-                headerActions={headerActions}
+                size={columns === 2 || columns === 4 ? 's' : 'm'}
```

Comment:
Nit: this can be a variable and reused, maybe call it columnsCount/Length so we know that it's not the actual columns